SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc

```diff
@@ -0,0 +1,454 @@
+    const G4ParticleDefinition*, G4double E0, G4double Z, G4double A, G4double cut, G4double)
+// Calculates the cross section per atom in GEANT4 internal units. Uses WW approximation to find the total cross section, performing numerical integrals over x and theta.
+{
+  G4double cross = 0.0;
+  if (E0 < keV || E0 < cut) {
+    return cross;
+  }
+
+  E0 = E0 / CLHEP::GeV;  //Change energy to GeV.
+  //G4double Mel = 5.1E-04;
+  G4double Mmu = 0.106;
+  if (E0 < 2. * MA)
+    return 0.;
+
+  //begin: chi-formfactor calculation
+  gsl_integration_workspace* w = gsl_integration_workspace_alloc(1000);
+
+  G4double result, error;
+  G4double tmin = MA * MA * MA * MA / (4. * E0 * E0);
+  G4double tmax = MA * MA;
+
+  gsl_function F;
+  ParamsForChi alpha = {1.0, 1.0, 1.0, 1.0};
+  F.function = &G4muDarkBremsstrahlungModel::chi;
+  F.params = &alpha;
+  alpha.AA = A;
+  alpha.ZZ = Z;
+  alpha.MMA = MA;
+  alpha.EE0 = E0;
+
+  //Integrate over chi.
+  gsl_integration_qags(&F, tmin, tmax, 0, 1e-7, 1000, w, &result, &error);
+
+  G4double ChiRes = result;
+  gsl_integration_workspace_free(w);
+
+  //Integrate over x. Can use log approximation instead, which falls off at high A' mass.
+  gsl_integration_workspace* dxspace = gsl_integration_workspace_alloc(1000);
+  gsl_function G;
+  G.function = &DsigmaDx;
+  G.params = &alpha;
+  G4double xmin = 0;
+  G4double xmax = 1;
+  if ((Mmu / E0) > (MA / E0))
+    xmax = 1 - Mmu / E0;
+  else
+    xmax = 1 - MA / E0;
+  G4double res, err;
+
+  gsl_integration_qags(&G, xmin, xmax, 0, 1e-7, 1000, dxspace, &res, &err);
+
+  G4double DsDx = res;
+  gsl_integration_workspace_free(dxspace);
+
+  G4double GeVtoPb = 3.894E08;
+  G4double alphaEW = 1.0 / 137.0;
+  G4double epsilBench = 1;
+
+  cross = GeVtoPb * 4. * alphaEW * alphaEW * alphaEW * epsilBench * epsilBench * ChiRes * DsDx * CLHEP::picobarn;
+  if (cross < 0.) {
+    cross = 0.;
+  }
+  E0 = E0 * CLHEP::GeV;
+
+  cross = cross * cxBias;
+  return cross;
+}
+
+G4DataVector* G4muDarkBremsstrahlungModel::ComputePartialSumSigma(const G4Material* material,
+                                                                  G4double kineticEnergy,
+                                                                  G4double cut)
+
+// Build the table of cross section per element.
+//The table is built for MATERIALS.
+// This table is used by DoIt to select randomly an element in the material.
+{
+  G4int nElements = material->GetNumberOfElements();
+  const G4ElementVector* theElementVector = material->GetElementVector();
+  const G4double* theAtomNumDensityVector = material->GetAtomicNumDensityVector();
+  G4DataVector* dv = new G4DataVector();
+  G4double cross = 0.0;
+
+  for (G4int i = 0; i < nElements; i++) {
+    cross += theAtomNumDensityVector[i] *
+             ComputeCrossSectionPerAtom(
+                 particle, kineticEnergy, (*theElementVector)[i]->GetZ(), (*theElementVector)[i]->GetA(), cut);
```

Comment:
@revering , better now than never. (*theElementVector)[i]->GetA() returns molar mass of the atom (see https://geant4.kek.jp/lxr/source/materials/include/G4Element.hh). You need to change to
(*theElementVector)[i]->GetN(). The same patch should be added in 12_4 and 12_5.